app/src/main/java/com/github/sdpteam15/polyevents/model/entity/UserEntity.kt

```diff
@@ -42,41 +41,51 @@ data class UserEntity(
             if (!value)
                 userProfiles
         }
+    var isLoading = false
     var userProfiles: ObservableList<UserProfile> = ObservableList()
         get() {
-            if (!loadSuccess)
-                Database.currentDatabase.userDatabase!!.getUserProfilesList(field, this)
-                    .observeOnce {
-                        if (it.value)
-                            loadSuccess = true
+            synchronized(this) {
+                if (!loadSuccess && !isLoading) {
+                    isLoading = true
+                    Database.currentDatabase.userDatabase!!.getUserProfilesList(field, this)
+                        .observeOnce {
+                            synchronized(this) {
+                                isLoading = false
+                                if (it.value && !loadSuccess)
+                                    loadSuccess = true
+                            }
+                        }
+                    field.observe {
+                        roles.clear(this)
+                        val res = mutableSetOf(UserRole.PARTICIPANT)
+                        for (e in it.value)
+                            res.add(e.userRole)
+                        roles.addAll(res, this)
                     }
+                }
+            }
```

Comment:
Nice use of synchronized. But why do we need on an instance of UserEntity?

Reply:
If we edit a profile the User Entity have to invalidate the cached value in profile liste and request them to the db if necessary